src/maidsafe/launcher/ui/models/account_handler_model.cc

```diff
@@ -24,17 +25,33 @@ namespace launcher {
 
 namespace ui {
 
-namespace models {
-
-AccountHandlerModel::AccountHandlerModel(QObject* parent)
-    : QObject{parent} {}
-
-}  // namespace models
+AccountHandlerModel::AccountHandlerModel(QObject* parent) : QObject{parent} {}
+
+AccountHandlerModel::~AccountHandlerModel() = default;
+
+std::unique_ptr<Launcher> AccountHandlerModel::Login(const QString& /*pin*/,
+                                                     const QString& /*keyword*/,
+                                                     const QString& /*password*/) {
+  on_scope_exit sig{[this] { emit LoginResultAvailable(); }};
```

Comment:
Yes this is on purpose to invoke a queued connection to Qt's GUI thread so that it can ".get()" the future to either receive the value or the exception